hbase-server/src/test/java/org/apache/hadoop/hbase/master/TestSplitWALManager.java

```diff
@@ -131,14 +131,14 @@ public void testWALArchiveWithDifferentWalAndRootFS() throws Exception{
     }
     executor.submitProcedure(splitProcedure);
     LOG.info("Submitted SplitProcedure.");
-    test_util_2.waitFor(30000, () -> executor.getProcedures().stream()
-      .filter(p -> p instanceof TransitRegionStateProcedure)
-      .map(p -> (TransitRegionStateProcedure) p)
-      .anyMatch(p -> TABLE_NAME.equals(p.getTableName())));
-    test_util_2.getMiniHBaseCluster().killRegionServer(
-      test_util_2.getMiniHBaseCluster().getRegionServer(0).getServerName());
-    test_util_2.getMiniHBaseCluster().startRegionServer();
-    test_util_2.waitUntilNoRegionsInTransition();
+    testUtil2.waitFor(30000, () -> executor.getProcedures().stream()
+      .filter(p -> p instanceof SplitTableRegionProcedure)
+      .map(p -> (SplitTableRegionProcedure) p)
+      .anyMatch(p -> TABLE_NAME.equals(p.getTableName())) && splitProcedure.isSuccess());
+    testUtil2.getMiniHBaseCluster().killRegionServer(
+      testUtil2.getMiniHBaseCluster().getRegionServer(0).getServerName());
+    testUtil2.getMiniHBaseCluster().startRegionServer();
+    testUtil2.waitUntilNoRegionsInTransition();
```

Comment:
Test looks fine, functionally wise, but could we simplify it and make it more clear by directly testing the log splitting? Sorry, don't want to be picky, it just looks we are doing more than needed here, maybe it would be simpler to just to do same thing as `testSplitLogs()`, after making sure wal is under a different file system. You could move current `testSplitLogs()` to a separate method that would be called by both this test and `testSplitLogs`. 

WDYT, @dasanjan1296 ?

Reply:
Yes, I agree with you. Simply doing a split wal operation should be enough to test this particular scenario. I went overboard with the test while trying to explore a few things. I've updated the test per your suggestion. PTAL.